src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts

```diff
@@ -226,6 +215,23 @@ export class CodeEditorTutorAssessmentContainerComponent implements OnInit, OnDe
         }
     }
 
+    private async onSubmissionReceived(submissionId: string, submission?: ProgrammingSubmission) {
+        if (!submission) {
+            // there are no unassessed submissions
+            this.submission = submission;
+            return;
+        }
+
+        // validate feedback here already so that overrides are possible for assessment note changes
+        // without touching the feedbacks
+        await this.handleReceivedSubmission(submission).then(() => this.validateFeedback());
```

Comment:
Nothing in the `onInit` actually depends on the `validateFeedback` method. This was just something I added so that valid feedback is actually recognized as such when the page is opened in the Client. So yes, both of your statements are right :D